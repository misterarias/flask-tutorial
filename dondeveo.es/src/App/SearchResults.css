.search-results {
  background: #cebada;
}

.img-circle {
  /* make it responsive */
  max-width: 100%;
  width: 100%;
  /* height: 100%; */
  display: block;

  /* make it a circle */
  padding: 50%;
  background: #e1e1e1;
  border-radius: 50%;

  /* Centering on image`s center*/
  background-position-y: center;
  background-position-x: center;
  background-repeat: no-repeat;

  /* it makes the clue thing, takes smaller dimension to fill div */
  background-size: cover;

  /* it is optional, for making this div centered in parent*/
  margin: 0 auto;
  top: 0;
  left: 0;
  right: 0;
  bottom: 0;
}
