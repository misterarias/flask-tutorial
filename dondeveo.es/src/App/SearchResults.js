import React, { Component } from 'react';
import './SearchResults.css';

class Movie extends Component {
  render() {
  return(
    <div className="result" onClick={() => this.props.handleClick(this.props.name)}>
      <h3>{this.props.name}</h3>
      <p>{this.props.description}</p>
    </div>
  );
  }
}

export default class SearchResults extends Component {

  constructor(props) {
    super();

    this.state = {
      results: props.results || []
    };
  }

  render() {
    const results = this.state.results.map( (item, key) => {
      return (
        <li key={key}>
          <Movie
            name={item}
            handleClick={this.props.handleClick}
            description="Created in a map()"
          />
        </li>
      );
    });

    return (
      <div className="search-results">
        <ul id="results">{results}</ul>
      </div>
    );
  }
}
