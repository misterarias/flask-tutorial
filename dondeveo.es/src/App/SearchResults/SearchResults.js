import React, { Component } from 'react';
import { Media, Row, Col, ListGroup, ListGroupItem, ListGroupItemHeading, ListGroupItemText } from 'reactstrap';
import ApiInfo from '../Api/ApiUtils' ;
import './SearchResults.css';

import { Route } from 'react-router-dom' ;

class Movie extends Component {
  render() {
    const movie = this.props.movie;

    let year = this.props.movie.year ;
    if (year.length > 4) {
      year = year.substr(0, 4);
    }

    let thumbnail = movie.thumbnail;
    if (thumbnail) {
      thumbnail = new ApiInfo().getImagePath(thumbnail);
    }

    return <Route render={ ({history}) => (
      <ListGroupItem
        action={true}
        className="listItem"
        onClick={ () => { history.push(`/movie/${movie.name}`); } }
      >
        <Row>
          <Col sm={2}>
            <Media object src={thumbnail} className="img-circle" />
          </Col>
          <Col sm={10}>
            <ListGroupItemHeading>{movie.name} <i className="small">({year})</i></ListGroupItemHeading>
            <ListGroupItemText>{movie.description}</ListGroupItemText>
          </Col>
        </Row>

      </ListGroupItem>
    )
    } />
  }
}

class SearchResults extends Component {


  render() {
    const results = this.props.results || [] ;
    const movieList = results.map( (item, key) => {
      return (
        <Movie
          key={key}
          movie={item}
        />
      );
    });

    return (
      <Row>
        <Col sm={12}>

          <p className="lead">
            { results.length > 0 ?`Mostrando ${results.length} resultados` : '' }
          </p>

          <ListGroup className="search-results">
            {movieList}
          </ListGroup>

        </Col>
      </Row>
    );
  }
}

export default SearchResults;
