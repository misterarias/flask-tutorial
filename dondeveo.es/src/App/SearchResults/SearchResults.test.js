'use strict'

import React from 'react';
import ReactDOM from 'react-dom';
import SearchResults from './SearchResults';

import { MemoryRouter, Route } from 'react-router-dom'

import {configure, shallow, mount} from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
configure({ adapter: new Adapter() })

import renderer from 'react-test-renderer';

const testMovieList = [
  {
    name: "Movie 1",
    description: "The best movie ever",
    year: '2010-01-01'
  },
  {
    name: "Movie 2",
    description: "Much better than Movie 1",
    year: '2017',
    thumbnail: 'real_path.jpg'
  }
];

it('renders correctly when empty', () => {
  const tree = renderer.create(
    <MemoryRouter >
      <SearchResults />
    </MemoryRouter>
  ).toJSON();

  expect(tree).toMatchSnapshot();
});


it('renders correctly with results', () => {
  const tree = renderer.create(
    <MemoryRouter >
      <SearchResults
        results={testMovieList}
      />
    </MemoryRouter>
  ).toJSON();

  expect(tree).toMatchSnapshot();
});

test('SearchResults starts with an empty list', () => {
  const searchResults = mount(
    <MemoryRouter >
      <SearchResults />
    </MemoryRouter>
  );

  const results = searchResults.find('ul').children();
  expect(results.length).toBe(0) ;
});

test('SearchResults renders results', () => {

  const searchResults = mount(
    <MemoryRouter >
      <SearchResults results={testMovieList}/>
    </MemoryRouter>
  );

  const renderedMovieList = searchResults.find('ul').children();
  expect(renderedMovieList.length).toBe(2) ;
  expect(renderedMovieList.first().find('h5').text()).toBe('Movie 1 (2010)');
  expect(renderedMovieList.last().find('h5').text()).toBe('Movie 2 (2017)');
});

test('SearchResults renders Routers for each link', () => {

  const searchResults = mount(
    <MemoryRouter >
      <SearchResults results={testMovieList} />
    </MemoryRouter>
  );

  const movieList = searchResults.find(Route);
  expect(movieList.length).toBe(2);
});
