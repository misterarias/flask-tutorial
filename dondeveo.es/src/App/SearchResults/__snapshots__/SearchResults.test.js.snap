// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly when empty 1`] = `
<div
  className="row"
>
  <div
    className="col col-sm-12"
  >
    <p
      className="lead"
    >
      
    </p>
    <ul
      className="search-results list-group"
    />
  </div>
</div>
`;

exports[`renders correctly with results 1`] = `
<div
  className="row"
>
  <div
    className="col col-sm-12"
  >
    <p
      className="lead"
    >
      Mostrando 2 resultados
    </p>
    <ul
      className="search-results list-group"
    >
      <li
        className="listItem list-group-item-action list-group-item"
        onClick={[Function]}
      >
        <div
          className="row"
        >
          <div
            className="col col-sm-2"
          >
            <img
              className="img-circle media-object"
              src={undefined}
            />
          </div>
          <div
            className="col col-sm-10"
          >
            <h5
              className="list-group-item-heading"
            >
              Movie 1
               
              <i
                className="small"
              >
                (
                2010
                )
              </i>
            </h5>
            <p
              className="list-group-item-text"
            >
              The best movie ever
            </p>
          </div>
        </div>
      </li>
      <li
        className="listItem list-group-item-action list-group-item"
        onClick={[Function]}
      >
        <div
          className="row"
        >
          <div
            className="col col-sm-2"
          >
            <img
              className="img-circle media-object"
              src="http://image.tmdb.org/t/p/w90/real_path.jpg"
            />
          </div>
          <div
            className="col col-sm-10"
          >
            <h5
              className="list-group-item-heading"
            >
              Movie 2
               
              <i
                className="small"
              >
                (
                2017
                )
              </i>
            </h5>
            <p
              className="list-group-item-text"
            >
              Much better than Movie 1
            </p>
          </div>
        </div>
      </li>
    </ul>
  </div>
</div>
`;
